meedan/check-web
# Feature/cv2-4502 articles trash action

- [x] Create button, add to edit slideout
- [x] Attach confirm proceed dialog, add copy from Brian
  - [x] Fact check: include link and title for media item
  - [x] Explainer: Only static message
- [x] Get data for attached media items, if the article has them (new data piece? existing data?)
- [x] Hook up Proceed to new trash API 
- [x] After user confirms trash: 
   - [x] Slideout is closed
   - [x] List of articles is updated to no longer show article
   - [x] Media items that were linked to the article no longer show article
   - [x] Number of articles goes down
## Description

Added a Move to Trash button that toggles the new trashed setting for articles.  If the article is active, when the user selects Edit, they have an option to move the article to the trash.  This will trigger a reload of the page at the moment, with proper Relay updates to come later. If the article is marked as trashed, they will now see a Restore from Trash button with a new Undo icon.  

## Type of change


## How has this been tested?

Manual testing:
- verified button only appears in Edit mode
- Trashed Fact Check with Media item and verified article is no longer associated, cannot find Article in search
- Trashed Explainer linked to Media Item and verified article is no longer associated, cannot find Article in search
- Trashed Explainer without linked Media, no longer appears in list or search
- Trashed Fact Check without media item, no longer appears in list or search
- After refresh, article count decreases by one
- After refresh, the list of Explainers/Fact-Checks no longer contains that article.  

## Things to pay attention to during code review

Restore from Trash and Move to Trash are the same button but with different settings. Right now, without a list of Trashed items, the Restore from Trash can't quite be tested. This portion was also listed as a part of the List feature, so the testing of that side can be saved until the list is in place and the trashed articles can be viewed.  The page refresh is a temporary solution until the Relay changes are made, but this will be handled in a separate ticket.  

## Checklist

- [X] If I touched a file that included an eslint-disable-file header, I updated the code such that the disabler can be removed
